python/cinder-understack/cinder_understack/dynamic_netapp_driver.py

```diff
@@ -328,11 +328,22 @@ def _volume_to_library(self, volume) -> Generator[NetAppMinimalLibrary]:
         original_host = volume["host"]
         # svm plus pool_name
         svm_pool_name = volume_utils.extract_host(original_host, level="pool")
+        if not svm_pool_name:
+            raise exception.InvalidInput(
+                reason=f"svm_pool_name not found in {original_host}"
+            )
+
         svm_name = svm_pool_name.split(_SVM_NAME_DELIM)[0]
+        # workaround when the vol_ name is detected instead of svm name
+        if svm_name.startswith("vol_"):
+            svm_name = f"os-{svm_name[4:]}"
```

Comment:
There's a config option for what the SVM name should be prefixed with. The check should probably be if the svm_name does not start with that prefix.